src/easylink/pipeline.py

```diff
@@ -1,3 +1,14 @@
+"""
+========
+Pipeline
+========
+
+This module is responsible for the ``Pipeline`` class, whose primary purpose
+is to perform pipeline-level validations as well as generate the Snakefile
```

Comment:
I'm not sure if there is a better way to explain this but I would try to avoid using pipeline (pipeline level validation) to describe what the pipeline class does.

Reply:
I think the purpose of the class is to generate a snakefile for the pipeline. Validation is just what is done to make sure it works as intended.

Reply:
I was trying to distinguish between high level validations (at the pipeline level) and other types (like at the implementation level). But I agree and think just removing "pipeline-level" is adequate.